demo-app/src/main/java/io/opentelemetry/android/demo/OtelDemoApplication.kt

```diff
@@ -37,19 +38,33 @@ class OtelDemoApplication : Application() {
                 .setDiskBufferingConfig(diskBufferingConfig)
 
         // 10.0.2.2 is apparently a special binding to the host running the emulator
+
+        val httpExporter = OtlpHttpSpanExporter.builder()
+            .setEndpoint("http://10.0.2.2:4318/v1/traces")
+            .setHeaders { mapOf("foo" to "bar") }
+            .build()
+        val swappableExporter = SwappableExporter(httpExporter)
         try {
-            rum = OpenTelemetryRumInitializer.initialize(
-                application = this,
-                endpointBaseUrl = "http://10.0.2.2:4318",
-                rumConfig = config
-            )
+            rum = OpenTelemetryRumBuilder.create(this, config)
+                .addSpanExporterCustomizer {
+                    swappableExporter
+                }
+                .build()
```

Comment:
In order to use this customized exporter approach, you cannot use the `OpenTelemetryRumInitializer` but instead must build the rum instance using the `OpenTelemetryRumBuilder`. This requires considerably more effort and care, and even as shown, does not correctly wire up the logging (events) exporter or even all the instrumentation you might need.